src/System.Web.OData/OData/Formatter/EdmLibHelpers.cs

```diff
@@ -825,12 +831,10 @@ public static IEnumerable<IEdmStructuralProperty> GetConcurrencyProperties(this
                 }
             }
 
-            if (_concurrencyProperties == null)
+            if (results.Any())
```

Comment:
Don't you want to cache the results whether they are empty or not, so that you don't have to check again next time you see the type?

Reply:
Thanks. I missed that.